src/Nethermind/Nethermind.Mev/Execution/TxBundleSimulator.cs

```diff
@@ -51,36 +57,67 @@ public Task<SimulatedMevBundle> Simulate(MevBundle bundle, BlockHeader parent, C
             }
             catch (OperationCanceledException)
             {
-                return Task.FromResult(new SimulatedMevBundle(bundle, false, 0, UInt256.Zero, UInt256.Zero));
+                return Task.FromResult(new SimulatedMevBundle(bundle, 
```

Comment:
Maybe some named method like SimulatedBundle.Cancelled(bundle) ? Also unnecessary allocations of List and Dictionary